### PR TITLE
Webhooks optimization

### PR DESCRIPTION
**Pull request by lkostrowski**
- [x] tests
- [x] react-query - to dynamically update webhooks state
- [x] validate tokens during webhooks-status fetch
- [x] use tokens from webhook instead fetching it

Zrzut ekranu 2023-04-27 o 16 52 28
Zrzut ekranu 2023-04-27 o 16 50 45

### CODE REVIEW

File: apps/search/src/domain/WebhookActivityToggler.service.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  service: "WebhookActivityTogglerService",
+});
+
+export interface IWebhooksClient {
+  fetchAppWebhooksIDs(id: string): Promise<string[]>;
+  disableSingleWebhook(id: string): Promise<void>;
+  enableSingleWebhook(id: string): Promise<void>;
+}
+
+export class WebhooksClient implements IWebhooksClient {
```
Comment (peelar) [inline apps/search/src/domain/WebhookActivityToggler.service.ts:19]: Nitpick: `WebhooksClient` is quite a generic name for an exported class. Given the name of the class below, I would name it something like `WebhooksActivityClient`.

File: apps/search/src/domain/WebhookActivityToggler.service.ts
Comment (peelar) [inline apps/search/src/domain/WebhookActivityToggler.service.ts:1]: Do you see any room for future generalization of the code and moving it to `shared`?

**Review by peelar — APPROVED** [review]

File: apps/search/src/domain/WebhookActivityToggler.service.ts
Comment (lkostrowski) [reply apps/search/src/domain/WebhookActivityToggler.service.ts:1]: Yes, but we need to figure out our approach to extract graphQL queries to shared. Maybe we need to extract codegen and generate such generic queries - we need some POC for that , cc @krzysztofwolski

File: apps/search/src/domain/WebhookActivityToggler.service.ts (hunk 1 above)
Comment (lkostrowski) [reply apps/search/src/domain/WebhookActivityToggler.service.ts]: improved the name

**Pull request merged**